pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -100,54 +115,156 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
+	if instance.Spec.Replicas != currentCountOfWindowsVMs {
+		// TODO: We're are swallowing the error which is a bad pattern, let's clean this up
+		if !r.reconcileWindowsNodes(instance.Spec.Replicas, currentCountOfWindowsVMs) {
+			return reconcile.Result{}, errors.New("error reconciling, requeuing again")
+		}
 	}
 
-	// Pod already exists - don't requeue
-	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
 	return reconcile.Result{}, nil
 }
 
-// newPodForCR returns a busybox pod with the same name/namespace as the cr
-func newPodForCR(cr *wmcv1alpha1.WindowsMachineConfig) *corev1.Pod {
-	labels := map[string]string{
-		"app": cr.Name,
+// reconcileWindowsNodes reconciles the Windows nodes so that required number of the Windows nodes are present in the
+// cluster
+func (r *ReconcileWindowsMachineConfig) reconcileWindowsNodes(desired, current int) bool {
+	// TODO: This can be decomposed further.
+	// 		TIP: Have a single method with operation like deletion, creation as argument
+	if desired < current {
+		hasDeletionFailed := r.deleteWindowsVMs(current - desired)
+		//r.reconcileCredentials()
+		return hasDeletionFailed
+	} else if desired > current {
+		// Let's not requeue the result for now. We can get the list of errors
+		hasCreationFailed := r.createWindowsVMs(desired - current)
+		//r.reconcileCredentials()
+		return hasCreationFailed
 	}
-	return &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      cr.Name + "-pod",
-			Namespace: cr.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "busybox",
-					Image:   "busybox",
-					Command: []string{"sleep", "3600"},
-				},
-			},
-		},
+	// TODO: If properly done, we should reconcile for creds here :).
+	// 		TIP: We can fail on VM creation/deletion reconciliation still reach here to make it more readable
+	/*if err := r.reconcileCredentials(); err != nil {
+		// Requeue
+	}*/
+	// Ideally this should be unreachable :)
+	return false
+}
+
+// reconcileCredentials ensures that credentials for each worker node is present in the cluster so that we can
+// access the Windows VM via those credentials
+func (r *ReconcileWindowsMachineConfig) reconcileCredentials() error {
+	// The windowsMachineConfig object has items related to the Windows objects that we can use to validate
+	// the credentials.
+	// Get the credentials from the configmap for all the objects we have in the Windows VM, if not go and update
+	// the configmap.
+	// Add a watch for the secrets, configmap here as well.
+	return nil
+}
+
+// chooseRandomNode chooses one of the windows nodes randomly
+func chooseRandomNode(windowsVMs map[types.WindowsVM]bool) types.WindowsVM {
+	for windowsVM := range windowsVMs {
+		return windowsVM
+	}
+	return nil
+}
+
+// deleteWindowsVMs returns the instance IDs of the successfully created Windows VMs
```

Comment:
seems like the comment needs to be updated, the function is returning a bool and it logs the instance Id of successfully deleted VM.